src/classes/UTIL_OrgTelemetry_SObject_BATCH.cls

```diff
@@ -0,0 +1,119 @@
+/*
+    Copyright (c) 2018, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2018
+* @group Utilities
+* @description To prevent a timeout error counting Opportunity records using specific filters, this batch job
+* is built to be generic enough to be executed in different configurations to count whatever large object requires
+* counting. Stateful is used to hold the count variables and an Enum is used to define the mode the batch job
+* is executing in.
+*/
+public class UTIL_OrgTelemetry_SObject_BATCH implements Database.Batchable<SObject>, Database.Stateful {
+
+    /**
+     * @description BatchJobMode is to add support for other query types in this same batch job
+     * that aren't yet needed, but may be in the future
+     */
+    public enum TelemetryBatchJobMode {
+        RECURRING_DONATIONS
+    }
+
+    /** @description The job mode this particular batch iteration is executing in */
+    private UTIL_OrgTelemetry_SObject_BATCH.TelemetryBatchJobMode jobMode;
+
+    /** @description Stateful counter vars used for the different counting jobs */
+    private Integer countAllRDOpps = 0;
+    private Integer countOpenRDOpps = 0;
+
+    private static final String recurringDonationOpenStatusValue = System.Label.npe03.RecurringDonationOpenStatus;
```

Comment:
The counter vars have `RD` in the name, should this var be named `rdOpenStatus` instead (no need for `Value`)?